drivers/java/src/test/java/com/rethinkdb/RethinkDBTest.java

```diff
@@ -488,6 +486,85 @@ public void testSaveObjectIgnoreNullPropertiesNested() {
         assertEquals(pojoOneInnerSelected.get("notNullProperty"), pojoOneInner.getNotNullProperty());
     }
 
+    @Test
+    public void testTableSelect_convertStringToDate() {
+        Locale.setDefault(Locale.JAPAN);
+        _convertStringToDate("2016-03-16", new String[]{
```

Comment:
When convert db string to `Date`/`OffsetDateTime`/`LocalDateTime`/`LocalDate`,
Support following common used date format:

```
2016-03-16, 
2016-03-16T09:58:59, 
2016-03-16T09:58:59.007, 
2016-03-16T09:58:59.007+09:00, 
Sat Dec 30 09:58:59 JST 2016
```

Let these string can be converted to any of above date types.